File: erykandroidcommon/src/main/java/pl/edu/amu/wmi/erykandroidcommon/recycler/select/SelectItemViewHolder.java
package pl.edu.amu.wmi.erykandroidcommon.recycler.select;

import android.view.View;
import android.widget.RadioButton;

import butterknife.BindView;
import butterknife.ButterKnife;
import lombok.Getter;
import pl.edu.amu.wmi.erykandroidcommon.R2;
import pl.edu.amu.wmi.erykandroidcommon.recycler.AbstractViewHolder;

/**
 * @author Eryk Mariankowski <eryk.mariankowski@247.codes> on 31.08.17.
 */
public abstract class SelectItemViewHolder<T> extends AbstractViewHolder<T> {

    @BindView(R2.id.radio)
    @Getter
    public RadioButton radio;

    public SelectItemViewHolder(View itemView) {
        super(itemView);
        ButterKnife.bind(this, itemView);
    }

    public abstract void bind(T item);
}

File: erykandroidcommon/src/main/java/pl/edu/amu/wmi/erykandroidcommon/recycler/select/SelectItemViewHolder.kt
package pl.edu.amu.wmi.erykandroidcommon.recycler.select

import android.view.View
import android.widget.RadioButton

import butterknife.BindView
import butterknife.ButterKnife
import lombok.Getter
import pl.edu.amu.wmi.erykandroidcommon.R2
import pl.edu.amu.wmi.erykandroidcommon.recycler.AbstractViewHolder

/**
 * @author Eryk Mariankowski <eryk.mariankowski></eryk.mariankowski>@247.codes> on 31.08.17.
 */
abstract class SelectItemViewHolder<T>(itemView: View) : AbstractViewHolder<T>(itemView) {

    @BindView(R2.id.radio)
    @Getter
    var radio: RadioButton? = null

    init {
        ButterKnife.bind(this, itemView)
    }

    abstract fun bind(item: T)
}
